src/rocdecode/vaapi/vaapi_videodecoder.h

```diff
@@ -88,6 +121,30 @@ class VaapiVideoDecoder {
     uint32_t num_slices_;
     VABufferID slice_data_buf_id_;
     uint32_t slice_data_buf_size_;
+
+    bool IsCodecConfigSupported(int device_id, rocDecVideoCodec codec_type, rocDecVideoChromaFormat chroma_format, uint32_t bit_depth_minus8, rocDecVideoSurfaceFormat output_format);
+    rocDecStatus CreateDecoderConfig();
+    rocDecStatus CreateSurfaces();
+    rocDecStatus CreateContext();
+    rocDecStatus DestroyDataBuffers();
+};
+
+// The GpuVaContext singleton class providing access to the the GPU VA services
+class GpuVaContext {
```

Comment:
Suggest adding a new header file gpu_va_context.h and move both definition and declaration of all the functions in the same class.